crypto/evp/pmeth_lib.c

```diff
@@ -339,20 +189,14 @@ void EVP_PKEY_meth_free(EVP_PKEY_METHOD *pmeth)
 
 EVP_PKEY_CTX *EVP_PKEY_CTX_new(EVP_PKEY *pkey, ENGINE *e)
 {
-    return int_ctx_new(NULL, pkey, e, NULL, NULL, -1);
+    return evp_pkey_ctx_new_int(NULL, pkey, e, NULL, NULL, -1);
 }
 
 EVP_PKEY_CTX *EVP_PKEY_CTX_new_id(int id, ENGINE *e)
 {
-    return int_ctx_new(NULL, NULL, e, NULL, NULL, id);
+    return evp_pkey_ctx_new_int(NULL, NULL, e, NULL, NULL, id);
 }
 
-EVP_PKEY_CTX *EVP_PKEY_CTX_new_provided(OPENSSL_CTX *libctx,
-                                        const char *name,
-                                        const char *propquery)
-{
-    return int_ctx_new(libctx, NULL, NULL, name, propquery, -1);
-}
```

Comment:
I disagree with spreading related functions away from each other.  `EVP_PKEY_CTX_new*` are variants of the same thing, please keep them together.
(and no, I don't think that FIPS sectioning is good enough a reason for the division, it still makes for code that's harder to read)

Reply:
I think its a case of 'choose your poison'.
Moved it around.